src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -347,17 +351,21 @@ await foreach (var response in requests)
 
             return null;
         }
-#nullable disable
 
-        public override async Task<ProvideSemanticTokensResponse> ProvideSemanticTokensAsync(
-            ProvideSemanticTokensParams semanticTokensParams,
+        public override async Task<ProvideSemanticTokensResponse?> ProvideSemanticTokensRangeAsync(
+            ProvideSemanticTokensRangeParams semanticTokensParams,
             CancellationToken cancellationToken)
         {
             if (semanticTokensParams is null)
             {
                 throw new ArgumentNullException(nameof(semanticTokensParams));
             }
 
+            if (semanticTokensParams.Range is null)
```

Comment:
Since you've nullable enabled this, and its not a public type, can we just remove this, and the other, null checks?

Reply:
My understanding is that might hide mistakes when non-nullable enabled code calls this code, but I could be mis-informed (and would LOVE to get rid of all that boiler-plate).

Reply:
Oh good point!